### PR TITLE
Not raise `NewPeer`/`LostPeer` for ourselves

### PR DESCRIPTION
**Pull request by vinipsmaker**
This change is [Reviewable](https://reviewable.io/reviews/maidsafe/crust/616)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/connection.rs [hunk 1]
```diff
@@ -221,6 +221,10 @@ pub fn connect(peer_contact: StaticContactInfo,
                     match deserialise::<ListenerResponse>(&read_buf[..bytes_rxd]) {
                         Ok(ListenerResponse::Connect { our_info, their_info, pub_key, }) => {
                             let (their_info, our_info) = (our_info, their_info);
+                            let their_pub_key = pub_key;
```
Comment (madadam) [inline src/connection.rs]: Nitpick, but can you instead rename the `pub_key` variable on line 222 to `their_pub_key` and then you wouldn't need this line at all?

Comment (vinipsmaker) [inline src/connection.rs]: I don't like the idea of renaming the variable on the struct. This struct is serialized and sent over the wire. It has different meanings when it's sent and when it's received.

Comment (madadam) [inline src/connection.rs]: Sorry, you are right. For some reason I though that it's a tuple, not struct. My bad.

**Pull request merged**